lib/layer/data.py

```diff
@@ -1322,6 +1302,22 @@ def _load_strokemap_from_ora(self, elem, x, y, orazip=None, oradir=None):
         else:
             raise ValueError("either orazip or oradir must be specified")
 
+    def get_paintable(self):
+        """True if this layer currently accepts painting brushstrokes"""
+        return (
+            self.visible and
+            not self.locked and
+            self.branch_visible and
+            not self.branch_locked
+        )
+
+    def get_fillable(self):
+        """True if this layer currently accepts flood fill"""
```

Comment:
As I mentioned, I leave FloodFill as is (well, except this method - I'll remove this). I think fixing it is out of the scope of this PR. Such a fix will change its current behavior quite noticeable, and this requires some discussion. I think it's better to create a separate issue/PR.